src/dune_rules/file_binding.ml

```diff
@@ -30,41 +33,53 @@ let relative_path_starts_with_parent relative_path =
   | Some (first, _) -> String.equal first Filename.parent_dir_name
 ;;
 
+let warning =
```

Comment:
Oh I see that we can't define this inside the function because all warnings need to be known about by the parser so they can be disabled in a config. In that case I think this should be renamed and moved to `warning.ml`.